docs/infrastructure.md

```diff
@@ -0,0 +1,82 @@
+# Infrastructure
+
+## Domains
+
+**nextstrain.org** is [hosted on Heroku](#heroku).
+
+**data.nextstrain.org** is an AWS CloudFronted S3 bucket, [`nextstrain-data`](#nextstrain-data).
+
+**staging.nextstrain.org** is an AWS CloudFronted S3 bucket, [`nextstrain-staging`](#nextstrain-staging).
+
+**login.nextstrain.org** is used by our [AWS Cognito user pool](#cognito).
+
+## Heroku
+
+The production Heroku app is [`nextstrain-server`](https://dashboard.heroku.com/apps/nextstrain-server), which is part of a [Heroku app pipeline of the same name](https://dashboard.heroku.com/pipelines/38f67fc7-d93c-40c6-a182-501da2f89d9d).
+Deploys of `master` happen automatically after Travis CI tests are successful.
+
+A testing/staging app, `nextstrain-dev`, is also used.
+
+### Environment variables
+
+  - `SESSION_SECRET` must be set to a long, securely generated string.
+    It protects the session data stored in browser cookies.
+    Changing this will invalidate all existing sessions and forcibly logout people.
+
+  - `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are tied to the `nextstrain.org` AWS IAM user.
+    These credentials allow the backend web server limited access to private S3 buckets.
+
+  - `REDIS_URL` is provided by the Heroku Redis add-on.
+    It should not be modified directly.
+    Our [authentication handlers](../authn.js) rewrite it at server start to use a secure TLS connection.
+
+### Add-ons
+
+The [Heroku Redis](https://elements.heroku.com/addons/heroku-redis) add-on is attached to our `nextstrain-server` and `nextstrain-dev` apps.
+Redis is used to persistently store login sessions after authentication via [AWS Cognito](#cognito).
+A persistent data store is important for preserving sessions across deploys and regular dyno restarts.
+
+## AWS
+
+All resources are in the `us-east-1` region.
+If you don't see them in the AWS Console, make sure to check the region you're looking at.
+
+## S3 buckets
+
+### nextstrain-data
+
+Public.
+CloudFronted.
+Contains JSONs for our core builds, as well as a few narratives and the `nextstrain.yml` conda environment definition.
```

Comment:
Narratives live in a github repo, not the S3 bucket. I'd be supportive if we wanted to move them from that repo into the S3 bucket (and same for staging). This would improve consistency among sources.

Reply:
Ah, great catch! Thanks. I've corrected it.

Reply:
It felt too fragile to me to have narratives live as deletable files on S3. One option would be . to do what we've done with https://github.com/nextstrain/conda and version narratives on GitHub but serve from S3.

Reply:
I agree that narratives should be in version control. It doesn't much matter for now if they're served from GitHub directly or from published copies on S3, so I'm inclined to leave it as-is from GitHub.